### PR TITLE
added watch for v2 resources

### PR DESCRIPTION
**Pull request by xh3b4sd**

### CODE REVIEW

File: service/resource/legacyv1/metric.go [hunk 1]
```diff
@@ -19,7 +19,3 @@ var versionBundleVersionGauge = prometheus.NewGaugeVec(
 	},
 	[]string{"major", "minor", "patch"},
 )
-
-func init() {
-	prometheus.MustRegister(versionBundleVersionGauge)
-}
```
Comment (xh3b4sd) [inline service/resource/legacyv1/metric.go]: Meh.

File: service/service.go [hunk 2]
```diff
@@ -141,6 +151,7 @@ func New(config Config) (*Service, error) {
 
 func (s *Service) Boot() {
 	s.bootOnce.Do(func() {
-		s.customObjectFramework.Boot()
+		go s.CRDFramework.Boot()
+		go s.customObjectFramework.Boot()
```
Comment (xh3b4sd) [inline service/service.go]: Bam.

File: service/framework.go [hunk 3]
```diff
@@ -9,28 +9,296 @@ import (
...
+	"github.com/giantswarm/flannel-operator/service/resource/namespacev2"
 	"github.com/giantswarm/flannel-operator/service/resource/networkconfigv1"
+	"github.com/giantswarm/flannel-operator/service/resource/networkconfigv2"
 )
 
 const (
 	ResourceRetries uint64 = 3
 )
 
+func newCRDFramework(config Config) (*framework.Framework, error) {
```
Comment (xh3b4sd) [inline service/framework.go]: Bam.

File: service/framework.go [hunk 4]
```diff
@@ -9,28 +9,296 @@ import (
...
+
+		legacyResource, err = legacyv2.New(legacyConfig)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	var networkConfigResource framework.Resource
+	{
+		c := networkconfigv2.DefaultConfig()
```
Comment (xh3b4sd) [inline service/framework.go]: Bam.

File: service/framework.go [hunk 5]
```diff
@@ -9,28 +9,296 @@ import (
 
 	"github.com/cenkalti/backoff"
 	"github.com/coreos/etcd/client"
+	"github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1"
+	"github.com/giantswarm/apiextensions/pkg/clientset/versioned"
```
Comment (xh3b4sd) [inline service/framework.go]: Bam.

**Comment by xh3b4sd** [comment]
Successfully tested on geckon.

File: service/framework.go (hunk 5 above)
Comment (JosephSalisbury) [reply service/framework.go]: Bam!

File: service/framework.go (hunk 3 above)
Comment (JosephSalisbury) [reply service/framework.go]: Bam!

File: service/framework.go (hunk 4 above)
Comment (JosephSalisbury) [reply service/framework.go]: Bam!

File: service/resource/legacyv1/metric.go (hunk 1 above)
Comment (JosephSalisbury) [reply service/resource/legacyv1/metric.go]: Bam!

File: service/service.go (hunk 2 above)
Comment (JosephSalisbury) [reply service/service.go]: Bam!

File: service/resource/legacyv1/metric.go (hunk 1 above)
Comment (xh3b4sd) [reply service/resource/legacyv1/metric.go]: Man's not consistent.

Comment (JosephSalisbury) [reply service/resource/legacyv1/metric.go]: Bam!

**Pull request merged**